Implement optional default for override_tag (#125)
torchbox/django-pattern-library: docs/README.md

1.  Override your template tag with a mock implementation
2.  Define fake result for your tag in a `yaml` file

### Providing a default value for template tags

> To provide a default for a template tag, you need to provide a keyword argument default when overriding your tag.

Reviewer: At present, this won't work with assigning the ouptut of a tag to a variable like `{% example_tag foo as bar %}` (currently this requires you to set `target_var` in the YAML file). We should document that limitation here or fix it so it works, possibly by checking if the overridden tag is a `simple_tag`?

Author: In the interest of getting this functionality into the build I've documented the limitation and we can pickup this extended functionality later.

```
from pattern_library.monkey_utils import override_tag

override_tag(register, 'a_tag_name', default="https://potato.com")
```

> This default is used for any tag that's not passed it's own context allowing specificity for certain elements that need it but stops the tags breaking when they're not structural to the component.

Reviewer: Grammar/typos

```suggestion
This default is used for any tag that's not passed its own context, allowing specificity for those elements that need it while preventing the tag from breaking when it's not structural to the component.
```
